### PR TITLE
Add select all environments box when copying feature rules to environments

### PR DESCRIPTION
**Pull request by bryce-fitzsimons**
### Features and Changes

Adds this checkbox:
image

- Closes **(add link to issue here)**

### CODE REVIEW

**Comment by AtakanColak** [comment]
Thanks 👍

File: packages/front-end/components/Features/CopyRuleModal.tsx [hunk 1]
```diff
@@ -32,13 +32,29 @@ export default function CopyRuleModal({
...
+  const envs = filteredEnvironments.map((e) => e.id);
 
   const ruleTxt = rules.length === 1 ? "rule" : "rules";
 
   const [selectedEnvironments, setSelectedEnvironments] = useState<
     Record<string, boolean>
   >({});
 
+  const allToggled =
+    Object.values(selectedEnvironments).filter((v) => v).length === envs.length;
```
Comment (gazzdingo) [inline packages/front-end/components/Features/CopyRuleModal.tsx:44]: the filter doesn't look like it is doing anything here i think we can remove that
```suggestion
    const allToggled = Object.values(selectedEnvironments).length === envs.length;
```

Comment (gazzdingo) [reply packages/front-end/components/Features/CopyRuleModal.tsx:44]: wait do we even need object.values here?

Comment (bryce-fitzsimons) [reply packages/front-end/components/Features/CopyRuleModal.tsx:44]: nah it's needed, otherwise it includes items like `prod: false`

Comment (gazzdingo) [reply packages/front-end/components/Features/CopyRuleModal.tsx:44]: wait it does its a boolean value :)

**Review by gazzdingo — APPROVED** [review]
Nit: selected environments is a little misleading i thought that it would only contain environments that are selected not all environments with a boolean flag.

**Pull request merged**